pkg/daemon/update.go

```diff
@@ -1555,13 +1557,33 @@ func (dn *Daemon) writeFiles(files []ign3types.File) error {
 		}
 
 		// To allow writing of "empty" files we'll allow source to be nil
-		contents := &dataurl.DataURL{}
+		var contentsBytes []byte
 		if file.Contents.Source != nil {
 			var err error
-			contents, err = dataurl.DecodeString(*file.Contents.Source)
+			contents, err := dataurl.DecodeString(*file.Contents.Source)
 			if err != nil {
 				return err
 			}
+			if file.Contents.Compression != nil {
+				switch *file.Contents.Compression {
+				case "":
+					contentsBytes = contents.Data
+				case "gzip":
+					reader, err := gzip.NewReader(bytes.NewReader(contents.Data))
+					defer reader.Close()
+					if err != nil {
+						return err
+					}
+					contentsBytes, err = io.ReadAll(reader)
+					if err != nil {
+						return err
+					}
+				default:
+					return fmt.Errorf("unsupported compression type %q for file %q", *file.Contents.Compression, file.Path)
```

Comment:
Reading the ignition docs again https://coreos.github.io/ignition/configuration-v3_2/, am I correct in understanding that the compression specifies if the content is compressed, and we uncompress it before writing it out?

Reply:
Yes, that's correct.